src/attribute_set.rs

```diff
@@ -0,0 +1,34 @@
+use std::fmt;
+use std::borrow::Cow;
+
+use Attribute;
+
+#[derive(Clone, Debug)]
+pub struct AttributeSet<'a> {
```

Comment:
Do we need a separate file/mod for this? `attribute::AttributeSet` also looks fine to me... Or better -- rename `attribute.rs` to `attr.rs`.